frontend/public/components/configmap-and-secret-data.jsx

```diff
@@ -1,15 +1,83 @@
+import * as _ from 'lodash-es';
+
 import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import { Heading } from './utils';
+
+export const ConfigMapData = ({data}) => {
+	const dl = [];
+	Object.keys(data || []).sort().forEach(k => {
+		dl.push(<dt key={`${k}-k`}>{k}</dt>);
+		dl.push(<dd key={`${k}-v`}><pre className="co-pre-wrap">{data[k]}</pre></dd>);
+	});
+
+	return <dl>{dl}</dl>;
+};
+
+export class SecretData extends React.PureComponent {
+	constructor(props) {
+		super(props);
+		this.state = { showSecret: false };
+		this.toggleSecret = this.toggleSecret.bind(this);
+	}
 
-const ConfigMapAndSecretData = ({data, decode}) => {
-  decode = decode || (v => v);
+	toggleSecret() {
+		this.setState({ showSecret: !this.state.showSecret })
+	}
 
-  const dl = [];
-  Object.keys(data || []).sort().forEach(k => {
-    dl.push(<dt key={`${k}-k`}>{k}</dt>);
-    dl.push(<dd key={`${k}-v`}><pre className="co-pre-wrap">{decode(data[k])}</pre></dd>);
-  });
+	render() {
+		const { data, type, decode } = this.props;
+		const { showSecret } = this.state;
+		const dl = [];
+		Object.keys(data).sort().forEach(k => {
+			if (k === '.dockercfg' || k === ".dockerconfigjson") {
+				const decodedData = JSON.parse(decode(data[k]));
+				const serverAddresses = decodedData.auths ? _.keys(JSON.parse(decode(data[k])).auths) : _.keys(JSON.parse(decode(data[k])));
```

Comment:
I see `JSON.parse(decode(data[k]));` in three places. I think this can be simplified. Maybe something like...

```javascript
const serverAddresses = _.keys(_.get(decodedData, 'auths', decodedData));
```